generative_data_prep/processors/article_tokenizer.py

```diff
@@ -334,10 +334,31 @@ def tokenize(self, completion: str, prompt: Optional[str] = None, category_id: O
             if self.prompt_postfix:
                 prompt = prompt + self.prompt_postfix
             prompt_token_ids = self.tokenizer.encode(prompt)
+            # If there is an EOS token at the end of the prompt then remove it
+            if (
+                hasattr(self.tokenizer, "eos_token_id")
+                and prompt_token_ids[-1] == self.tokenizer.eos_token_id
+                and len(prompt_token_ids) >= 1
```

Comment:
Maybe also check that the completion is non-empty? There's an (admittedly unlikely) corner case where the completion is empty, but the prompt is (intentionally) non-empty. Perhaps we can add this kind of case to the tests as well.

Reply:
I think we should still remove, if you check a few lines below, we always add an EOS after every line. So if the tokenizer adds one in automatically, we should remove it before adding our default EOS

- Also keep in mind that by default prompt only sequences are removed in the `_remove_prompt_only_sequences` function

Reply:
Good catch, thank you!